linkml/generators/sqltablegen.py

```diff
@@ -238,73 +239,95 @@ def _sql_name(sn: str):
         schema_metadata.create_all(engine)
         return ddl_str
 
-    def get_sql_range(self, slot: SlotDefinition, schema: SchemaDefinition = None):
+    def get_oracle_sql_range(self, slot: SlotDefinition) -> Text | VARCHAR2 | None:
         """
-        returns a SQL Alchemy column type
+        Generate the appropriate range for Oracle SQL.
+
+        :param slot: the slot under examination
+        :type slot: SlotDefinition
+        :return: appropriate type for Oracle SQL or None
+        :rtype: Text | VARCHAR2 | None
         """
-        range = slot.range
+        slot_range = slot.range
+        if not slot_range:
+            return None
+
+        if slot_range.lower() in ["str", "string"]:
+            # string type data should be represented as a VARCHAR2
+            return VARCHAR2(self.default_length_oracle)
+
+        # check whether the slot range matches the regex "VARCHAR2?(\((\d+)\))?"
+        match = re.match(VARCHAR_REGEX, slot_range)
+        if match:
+            # match.group(2) is the digits in brackets after VARCHAR
+            # i.e. a defined length for the VARCHAR
+            if match.group(2) and int(match.group(2)) > ORACLE_MAX_VARCHAR_LENGTH:
+                msg = (
+                    "Warning: range exceeds maximum Oracle VARCHAR length, "
+                    f"CLOB type will be returned: {slot_range} for {slot.name} = {slot.range}"
+                )
+                logger.info(msg)
+                return Text()
+            # set the length to either the varchar length (as defined in the slot_range)
+            # or the default
+            return VARCHAR2(match.group(2) or self.default_length_oracle)
+
+        # use standard SQL range matching for anything else
+        return None
+
+    def get_sql_range(self, slot: SlotDefinition, schema: SchemaDefinition = None):
+        """Get the slot range as a SQL Alchemy column type."""
+        slot_range = slot.range
+
+        if self.dialect == "oracle":
+            range_type = self.get_oracle_sql_range(slot)
+            if range_type:
+                return range_type
+
+        if slot_range is None:
+            return Text()
+
         # if no SchemaDefinition is explicitly provided as an argument
         # then simply use the schema that is provided to the SQLTableGenerator() object
         if not schema:
-            schema = SchemaLoader(data=self.schema).resolve()
```

Comment:
SchemaLoader was throwing errors with the personinfo.yaml schema, and since SchemaView is used later in the function, just switch to using SV for accessing class / enum / type details.